src/lib/hooks/use-best-quote/utils/issuance/legacy-quote.ts

```diff
@@ -71,11 +71,11 @@ export async function getLegacyRedemptionQuote(
     const outputTokens = isIcReth
       ? [RETH]
       : components.map((component) => {
-          const outputToken = getTokenDataByAddress(component, chainId)!
+          const outputToken = getTokenByChainAndAddress(chainId, component)!
```

Comment:
Why not  `getChainComponentTokenList(chainId)`?

Reply:
Theoretically, this make sense but we may have indices here that actually hold other index tokens as components e.g. BED having DPI as component. So I'd say this is the safer option for now.

<img width="614" alt="Bildschirmfoto 2024-10-11 um 11 27 35" src="https://github.com/user-attachments/assets/37951029-adaf-45d0-af64-c948b50842f9">

Reply:
DPI has the index and the component tag so it will be returned : )